Source/JSONAPISerializer.swift

```diff
@@ -67,33 +69,52 @@ public protocol JSONAPIEntity: CustomSerializable, JSONAPISerializable {
     var id: String { get }
 }
 
+private typealias JSONAPIConvertible = protocol<JSONAPISerializable, Serializable>
+
+/// A wrapper struct that handles the serialization of the JSON API data field
+private struct JSONAPIDataWrapper: CustomSerializable {
+    let object: JSONAPIConvertible
+    
+    private func customSerialize(keyTransformer: KeyTransformer?) -> AnyObject? {
+        return object.data(includeRelationships: true, includeAttributes: true, keyTransformer: keyTransformer)
+    }
+}
+
+/// A wrapper struct that handles the serialization of the JSON API include field
+private struct JSONAPIIncludedWrapper: CustomSerializable {
+    let object: JSONAPIConvertible
+    let includeChildren: Bool
+    
+    private func customSerialize(keyTransformer: KeyTransformer?) -> AnyObject? {
+        return object.includedRelationships(includeChildren, keyTransformer: keyTransformer)?.unifiedIncludedRelationships()
+    }
+}
```

Comment:
We need something like this because JSONAPIEntity is a strange `Serializable`. JSONAPIEntity's serialize method just call ` object.data(includeRelationships ....` and `includedRelationships(...` is never in the serialized entity. 
`JSONAPIEntity` probably shouldn't even be a `Serializable` object (we might consider to remove the conformance).
Because of this I needed to handle the keyTransformer and I thought that the easiest and cleaner option would be to use `CustomSerializable` Wrappers that makes it really easy to handle the keyTransformer. 
Also now at initialization time nothing is serialized (while before we were creating the included field since it was not part of the serialization).